Source/Csla/Core/GraphMerger.cs

```diff
@@ -202,5 +213,155 @@ private void MergeGraph(IEditableCollection target, IEditableCollection source)
         target.Remove(item);
       GetDeletedList<C>(target).Clear();
     }
+
+    #region Async Methods
+    /// <summary>
+    /// Merges state from source graph into target graph.
+    /// </summary>
+    /// <param name="target">Target of merge.</param>
+    /// <param name="source">Source for merge.</param>
+    public async Task MergeGraphAsync(IEditableBusinessObject target, IEditableBusinessObject source)
+    {
+      if (target is IManageProperties imp)
+      {
+        FieldManager.FieldDataManager targetFieldManager = null;
+        if (target is IUseFieldManager iufm)
+          targetFieldManager = iufm.FieldManager;
+        FieldManager.FieldDataManager sourceFieldManager = null;
+        if (source is IUseFieldManager iufms)
+          sourceFieldManager = iufms.FieldManager;
+
+        var targetProperties = imp.GetManagedProperties();
+        foreach (var item in targetProperties)
+        {
+          var sourceFieldExists = true;
+          if (sourceFieldManager != null && (item.RelationshipType & RelationshipTypes.LazyLoad) == RelationshipTypes.LazyLoad)
+            sourceFieldExists = sourceFieldManager.FieldExists(item);
+          object sourceValue = null;
+          if (sourceFieldExists)
+            sourceValue = ReadProperty(source, item);
+          if (sourceValue is IEditableBusinessObject sourceChild)
+          {
+            var targetFieldExists = true;
+            if (targetFieldManager != null && (item.RelationshipType & RelationshipTypes.LazyLoad) == RelationshipTypes.LazyLoad)
+              targetFieldExists = targetFieldManager.FieldExists(item);
+            if (targetFieldExists && ReadProperty(target, item) is IEditableBusinessObject targetChild)
+            {
+              await MergeGraphAsync(targetChild, sourceChild);
+            }
+            else
+            {
+              if ((item.RelationshipType & RelationshipTypes.PrivateField) == RelationshipTypes.PrivateField)
+                Reflection.MethodCaller.CallPropertySetter(target, item.Name, sourceChild);
+              else
+               LoadProperty(target, item, sourceChild);
+            }
+          }
+          else
+          {
+            if (sourceValue is IEditableCollection sourceList)
+            {
+              var targetFieldExists = true;
+              if (targetFieldManager != null && (item.RelationshipType & RelationshipTypes.LazyLoad) == RelationshipTypes.LazyLoad)
+                targetFieldExists = targetFieldManager.FieldExists(item);
+              if (targetFieldExists && ReadProperty(target, item) is IEditableCollection targetList)
+              {
+                MergeGraph(targetList, sourceList,true);
```

Comment:
Doesn't this defeat the purpose? This is a sync call, and the modification to `MergeGraph` presumably does a sync call even to the methods ending in `Async`?

Reply:
My bad. I have changed the code.